org.eclipse.jdt.core/codeassist/org/eclipse/jdt/internal/codeassist/complete/CompletionOnQualifiedNameReference.java

```diff
@@ -77,10 +77,8 @@ public TypeBinding resolveType(BlockScope scope) {
 
 		throw new CompletionNodeFound();
 	}
-
-	return new CompletionNodeFound(this, this.binding, scope).throwOrDeferAndReturn(() -> {
-		// probably not in the position to do useful resolution, just provide some binding
-		return this.resolvedType = new ProblemReferenceBinding(this.tokens, null, ProblemReasons.NotFound);
-	});
+	
+	// probably not in the position to do useful resolution, just provide some binding
+	return new CompletionNodeFound(this, this.binding, scope).throwOrDeferAndReturn(() -> (this.resolvedType = new ProblemReferenceBinding(this.tokens, null, ProblemReasons.NotFound)));
```

Comment:
This change is wrong. The code comment lost the connection to its code.
@vogella please don't mess up our code!

Reply:
PR to move the comment back https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1695

Reply:
> PR to move the comment back #1695

No longer necessary, I had to recover it before I could fix #1440.

Reply:
I leave it to you as experienced JDT developer to decide if you want to report that incorrect behaviour of the JDT clean-up action to jdt UI.

Reply:
> I leave it to you as experienced JDT developer to decide if you want to report that incorrect behaviour of the JDT clean-up action to jdt UI.

That looks like a perfect task for a person who loves these clean-up actions. I am not that person.

This incident only confirmed my worries that those actions may suggest that JDT is responsible for the outcome, whereas responsibility can only be owned by a person, viz. the contributor proposing the resulting change.